solvation_analysis/analysis_library.py

```diff
@@ -0,0 +1,163 @@
+import pandas as pd
+import numpy as np
+
+
+class _SolvationData:
+    """
+    A class for holding solvation data, this will slightly reprocess data and
+    make it available in different forms.
+    # TODO: is there any way to enforce immutability? Users shouldn't mutate this!
+    """
+
+    def __init__(self, raw_solvation_frames):
+        """
+        Parameters
+        ----------
+        raw_solvation_frames: a list of tidy dataframes from the Solution class,
+            encapsulating all solvent input
+        """
+        self.raw_solvation_frames = raw_solvation_frames
```

Comment:
This makes sense to me. I think that I am going to just put **all** of the data in a dataframe rather than having a list of dataframes. That will allow analyses to filter by solute or by time step with the same notation. 

I'll stack them up as a numpy array and then put them to pandas at the end.

Reply:
Awesome. I think this will speed things up and also make `SolvationData` more encapsulated, ie it is the primary owner of the all important `DataFrame`

Reply:
After I make the change, I think `SolvationData` might be no longer needed. It currently holds logic to iterate through the list of frames, but once everything is in a single data frame, that won't be necessary. 

Might be simpler to just pass around the god `DataFrame` rather than wrapping into a class. Happy to discuss once I change things.

Reply:
"Simpler is better" — Python Zen

(I.e., in favor of dropping unnecessary wrapper classes if the data structure is already expressive enough.)

Reply:
`SolvationData` is gone and everything is in a giant data frame. Going to resolve this now!